jsapp/js/components/modalForms/dataAttachmentColumnsForm.es6

```diff
@@ -0,0 +1,226 @@
+import React from 'react';
+import autoBind from 'react-autobind';
+import MultiCheckbox from 'js/components/common/multiCheckbox';
+import {bem} from 'js/bem';
+import {actions} from 'js/actions';
+import {renderLoading} from '../modalForms/modalHelpers';
+
+/**
+ * Attributes from parent needed to generate `columnsToDisplay`
+ *
+ * @namespace parentAttributes
+ * @prop {string} uid
+ * @prop {string} name
+ * @prop {string} url
+ */
+
+/**
+ * The content of the DATA_ATTACHMENT_COLUMNS modal
+ *
+ * @prop {function} onSetModalTitle - for changing the modal title by this component
+ * @prop {function} onModalClose - causes the modal to close
+ * @prop {function} triggerParentLoading - triggers loading on parent modal
+ * @prop {function} generateColumnFilters - generates columns for multicheckbox
+ * @prop {object} asset - current asset
+ * @prop {parentAttributes} parent
+ * @prop {string} fileName
+ * @prop {string[]} fields - child selected fields
+ * @prop {string} attachmentUrl - if exists, we are patching an existing attachment
+                                  otherwise, this is a new import
+ */
+class dataAttachmentColumnsForm extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      isVirgin: true,
+      isLoading: false,
+      columnsToDisplay: [],
+    };
+    autoBind(this);
+    this.unlisteners = [];
+  }
+
+  componentDidMount() {
+    // We must query for parent's asset content in order to display their
+    // available columns
+    actions.resources.loadAsset({id: this.props.parent.uid});
+
+    this.unlisteners.push(
+      actions.dataShare.attachToParent.completed.listen(
+        this.onAttachToParentCompleted
+      ),
+      actions.dataShare.attachToParent.failed.listen(
+        this.onAttachToParentFailed
+      ),
+      actions.dataShare.patchParent.completed.listen(
+        this.onPatchParentCompleted
+      ),
+      actions.dataShare.patchParent.failed.listen(
+        this.onPatchParentFailed
+      ),
+      actions.resources.loadAsset.completed.listen(
+        this.onLoadAssetContentCompleted
+      ),
```

Comment:
Good catch :) if it failed then the loading would stay forever--doesn't make it clear enough that a request failed. I thought a notification was already being handled in `actions.es6`, but that's not true. Interestingly, nowhere else in the JS code (after a quick search) listens for `actions.resources.loadAsset.failed`, probably because a blank screen is a good enough indicator that something messed up :P